libslide/src/bignum/utils.rs

```diff
@@ -18,40 +49,42 @@ pub fn abs(rhs: &Bignum) -> Bignum {
     }
 }
 
+pub fn ispowerof2(n: f64) -> bool {
+    if n == 0.0 {
+        return false;
+    }
+    n.log2().ceil() - n.log2().floor() < TOLERANCE
+}
```

Comment:
Does it have to be an f64? If we can cast to an unsigned int, this is much better:

```
n != 0 && (n & (n-1) == 0)
```

Reply:
the log2 function only is available for f64 not usigned

Reply:
Yeah but you can replace the whole function with `n != 0 && (n & (n-1) == 0)`, then you dont need log2

Reply:
Also this comment